### PR TITLE
[Module update] Validatefomcomponents

### PR DESCRIPTION
**Pull request by WackerO**
This PR updates the validatefomcomponents module and adapts the workflow accordingly

Also closes https://github.com/nf-core/differentialabundance/issues/56

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/differentialabundance/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/differentialabundance _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/differentialabundance.nf [hunk 1]
```diff
@@ -73,46 +73,46 @@ def multiqc_report = []
...
     // Get a features table from the GTF and combine with the matrix and sample
     // annotation (fom = features/ observations/ matrix)
 
     GTF_TO_TABLE( file_gtf, [[ "id":""], []])
-    
-    // Combine features with the observations and matrices to create a FOM
-    // where these things can travel together
+    ch_feature_anno = GTF_TO_TABLE.out.feature_annotation.map{
+        tuple( exp_meta, it[1])
+    }
```
Comment (pinin4fjords) [inline workflows/differentialabundance.nf:96]:
```suggestion
    ch_feature_anno = GTF_TO_TABLE.out.feature_annotation
    .map{
        tuple( exp_meta, it[1])
    }
```
sorry, just a style consistency thing

File: workflows/differentialabundance.nf [hunk 2]
```diff
@@ -217,13 +214,15 @@ workflow DIFFERENTIALABUNDANCE {
         }
         .unique()
 
-    ch_all_matrices = VALIDATOR.out.fom
-        .combine(ch_processed_matrices)                         // Add processed marices to what we have in the FOM
+    ch_all_matrices = VALIDATOR.out.sample_meta
+        .combine(VALIDATOR.out.feature_meta.map{ it[1] })
+        .combine(VALIDATOR.out.assays.map{ it[1] })
+        .combine(ch_processed_matrices)
```
Comment (pinin4fjords) [inline workflows/differentialabundance.nf:220]:
```suggestion
    ch_all_matrices = VALIDATOR.out.sample_meta              // meta, samples
        .join(VALIDATOR.out.feature_meta)                                // meta, samples, features
        .join(VALIDATOR.out.assays)                                           // meta, samples, features, raw matrix
        .combine(ch_processed_matrices)                                // meta, samples, features, raw, norm, vst
```

No need for the combines and maps- just use a join (these channels have the same meta up to the differential part)

File: workflows/differentialabundance.nf [hunk 3]
```diff
@@ -259,16 +258,16 @@ workflow DIFFERENTIALABUNDANCE {
     ch_citations_file = Channel.from(citations_file)
 
     ch_report_input_files = ch_all_matrices
-        .tail()
+        .map{tuple(it[1], it[2], it[3])}
```
Comment (pinin4fjords) [inline workflows/differentialabundance.nf:261]:
```suggestion
        .map{ it.tail() }
```
I think I see what you were doing there (my .`.tail()` wasn't right, though it didn't break anything).

But this is a simpler way of removing the meta.

**Review by pinin4fjords — CHANGES_REQUESTED** [review]
Thanks @WackerO , it's much more readable with the new validate interface, appreciate your efforts there. 

I've added a couple of code-related fixes, and will be happy to approve a merge as soon as those are addressed and all the random space changes etc have been removed.

File: modules/nf-core/shinyngs/validatefomcomponents/main.nf [hunk 4]
```diff
@@ -2,18 +2,21 @@ process SHINYNGS_VALIDATEFOMCOMPONENTS {
     tag "$sample"
     label 'process_single'
 
-    conda "bioconda::r-shinyngs=1.5.1"
+    conda "bioconda::r-shinyngs=1.5.4"
```
Comment (pinin4fjords) [inline modules/nf-core/shinyngs/validatefomcomponents/main.nf:5]: Sorry, one last thing- would you be able to bump the other shinyngs modules ass well (exploratory, differential)?

Then this PR will also close https://github.com/nf-core/differentialabundance/issues/56.

File: modules.json [hunk 5]
```diff
@@ -45,6 +45,11 @@
                         "git_sha": "c8e35eb2055c099720a75538d1b8adb3fb5a464c",
                         "installed_by": ["modules"]
                     },
+                    "multiqc": {
```
[the remark this answers is not in the document]
Comment (WackerO) [reply modules.json:48]: Gosh, why does this keep being readded? I'll remove it once again...

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## v1.1.0dev - xxxx-xx-xx
 
+### `Changed`
+
+- [[#58](https://github.com/nf-core/differentialabundance/issues/58)] - Update module shinyngs/validatefomcomponents ([@WackerO](https://github.com/WackerO), review by [@pinin4fjords](https://github.com/pinin4fjords))
```
Comment (pinin4fjords) [inline CHANGELOG.md:10]:
```suggestion
- [[#57](https://github.com/nf-core/differentialabundance/issues/57)] - Update module shinyngs/validatefomcomponents and bump other shinyngs modules to resolve [[#56](https://github.com/nf-core/differentialabundance/issues/56)] ([@WackerO](https://github.com/WackerO), review by [@pinin4fjords](https://github.com/pinin4fjords))
```
Fixed the wrong issue number and added the other one

File: workflows/differentialabundance.nf [hunk 7]
```diff
@@ -217,10 +215,12 @@ workflow DIFFERENTIALABUNDANCE {
         }
         .unique()
 
-    ch_all_matrices = VALIDATOR.out.fom
-        .combine(ch_processed_matrices)                         // Add processed marices to what we have in the FOM
+    ch_all_matrices = VALIDATOR.out.sample_meta              // meta, samples
```
Comment (pinin4fjords) [inline workflows/differentialabundance.nf:218]: Could you align these comments? Apologies, the GitHub suggestions editor misaligned them.

**Review by pinin4fjords — APPROVED** [review]
If you just do those final two minor things before you merge that would be good, otherwise good work!

The full tests are failing locally with the full test data, but it's a minor fix I know how to do and I'll handle it separately.

**Pull request merged**